cmd/liqoctl/main.go

```diff
@@ -5,21 +5,32 @@ import (
 	"os"
 	"os/signal"
 	"syscall"
+	"time"
 
 	"github.com/spf13/cobra"
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 
 	liqocmd "github.com/liqotech/liqo/cmd/liqoctl/cmd"
 )
 
+const (
+	terminationTimeout = 5 * time.Second
+)
+
 func main() {
 	ctx, cancel := context.WithCancel(context.Background())
 	sig := make(chan os.Signal, 1)
 	signal.Notify(sig, syscall.SIGINT, syscall.SIGTERM)
 	go func() {
 		<-sig
 		cancel()
+		select {
+		case <-time.NewTimer(terminationTimeout).C:
+			os.Exit(1)
+		case <-ctx.Done():
+			return
```

Comment:
This should return immediately, since you canceled the context. One of the alternatives I saw (in the controller-runtime library IIRC) was close the signal and cause the exit if invoked a second time.